desc/compute/utils.py

```diff
@@ -91,16 +96,37 @@ def compute(parameterization, names, params, transforms, profiles, data=None, **
         data=data,
         **kwargs,
     )
+
+    # convert data from default 'rpz' basis to 'xyz' basis, if requested by the user
+    if basis == "xyz":
+        from .geom_utils import rpz2xyz, rpz2xyz_vec
+
+        for name in data.keys():
+            errorif(
+                data_index[p][name]["dim"] == (3, 3),
+                NotImplementedError,
+                "Tensor quantities cannot be converted to Cartesian coordinates.",
+            )
+            if data_index[p][name]["dim"] == 3:  # only convert vector data
+                if name == "x":
+                    data[name] = rpz2xyz(data[name])
+                else:
+                    data[name] = rpz2xyz_vec(data[name], phi=data["phi"])
+
     return data
 
 
-# TODO: can we do the basis change here possibly? instead of repeating it everywhere?
-#  Maybe give an "inherent basis" parameter to the compute decorator
-#  so we check what the natural basis is versus what the desired is?
 def _compute(
     parameterization, names, params, transforms, profiles, data=None, **kwargs
```

Comment:
maybe add a check that basis isnt in kwargs here? (it gets popped in the other function so wouldn't normally be passed in here)